NEWPART.md

```diff
@@ -0,0 +1,203 @@
+
+
+
+### Step 2 ###
+
+Add a new settings file.  Usually you will start with an existing settings file and modify it.
+Assuming you're in prjxray/,
+
+```
+cp settings/artix7_200t.sh settings/artix7_100t.sh
+git add settings/artix7_100t.sh
+```
+
+Update the following values in the new settings file:
+* `XRAY_PART` -- choose a package that is fully bonded (typically the largest number of pins)
+* `XRAY_ROI_TILEGRID` -- modify the bounding boxes to be a tight fit on your new part.
+* `XRAY_IOI3_TILES` -- See the comments in 005 fuzzers.  Special handling.  Ask Keith or Tim or Tim.
+* `XRAY_PIN_00` -- this must be a clock pin.   You can look at the device in the Vivado GUI interactively (click on an IO and look at its properties until you find one with IS\_CLOCK=true), or  run a small test in Vivado and see which pin assigned the 'clk'.
+* `XRAY_PIN_01` and on -- these should be normal data pins on the device.
+
+Source this new settings file:
+
+```
+source settings/artix7_100t.sh
+```
+
+### Step 3 ###
+
+Edit the top Makefile
+
+* Add the new part to the correct list.   This is the basename of the new settings file that you just created. In this case, we added `artix7_100t`:
+
+```
+ARTIX_PARTS=artix7_200t artix7_100t
+```
+
+
+### Step 4 ###
+
+Make the following target, adapted for your new part, and setting the parallelism factor appropriate for your machine:
+
+```
+make -j32 MAX_VIVADO_PROCESS=32 db-part-only-artix7_100t
+```
+
+* It should run fuzzers 000, 001, 005, 072, 073, 074, and 075.
+                
+* 005 will take a long time unless you throw a lot of cores at it.  32-64 cores suggested.
+
+* You _may_ get a spurious failure, since the parallel runs may violate ordering dependencies among the fuzzers.  If this happens, clean the fuzzer that failed (e.g.  `make -C fuzzers/005-tilegrid clean`), and then rerun the top-level make.
+
+* 074 _will_ fail the first time, since we haven't told it to ignore certain wires.
+    * Go to the build directory `cd fuzzers/074-dump_all/build_xc7a100tfgg676-1` (substitute your new part here and below)
+    * Run `python3 ../analyze_errors.py --output_ignore_list  > new-ignored`
+    * Inspect and compare against existing ignored wire files
+    * IF IT LOOKS GOOD, copy it to an appropriately-named file: `cp new-ignored ../ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt`
+    * Add it: `git add ../ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt`
+
+* Return to prjxray/ directory, and clean up 074 to prepare for the rerun: `make -C fuzzers/074-dump-all clean`
+
+* Rerun the top make command, `make -j32 MAX_VIVADO_PROCESS=32 db-part-only-artix7_100t`
+
+### Step 5 ###
```

Comment:
When this is committed, add a comment to https://github.com/SymbiFlow/prjxray/issues/1476 to remove this step once we have updated the part processing stuff.  The instructions here also need to be updated as part of https://github.com/SymbiFlow/prjxray/issues/1475 too

Reply:
Yes, good catch.   I'll leave this comment unresolved, and add a comment to those PRs when this doc's final location is determined.

Reply:
I also need to check if https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1791 affects my instructions (under "Database Updates")